armi/apps.py

```diff
@@ -260,13 +260,38 @@ def registerUserPlugins(self, pluginPaths):
         restrict their flexibility and power as compared to the usual ArmiPlugins.
         """
         for pluginPath in pluginPaths:
+            if self._isPluginRegistered(pluginPath):
+                continue
             if ".py:" in pluginPath:
                 # The path is of the form: /path/to/why.py:MyPlugin
                 self.__registerUserPluginsAbsPath(pluginPath)
             else:
                 # The path is of the form: armi.thing.what.MyPlugin
                 self.__registerUserPluginsInternalImport(pluginPath)
 
+    def _isPluginRegistered(self, pluginPath: str):
```

Comment:
I love the PR!

Do you think we could get a unit test of this new method?

If it helps, we already have a lot of other [UserPlugin unit tests](https://github.com/terrapower/armi/blob/main/armi/tests/test_user_plugins.py).

Or would you like a little help writing that test?

Reply:
I can take a look at that link and see about putting together a unit test. I wanted to see if this would be how you wanted to handle the problem :)

Reply:
You made a great test! Thanks!